web/src/integTest/java/com/microsoft/applicationinsights/azuretest/CompareTelemetryWithAzureExportTests.java

```diff
@@ -0,0 +1,342 @@
+     * @param json The JSON object
+     * @return A TelemetryItem
+     * @throws Exception
+     */
+    private TelemetryItem getTelemetryItemFromJson(DocumentType docType, JSONObject json) throws Exception {
+        if (docType == DocumentType.Requests) {
+            return getRequestTelemetryItemFromJson(json);
+        }
+
+        if (docType == DocumentType.PerformanceCounters) {
+            return getPerformanceCounterTelemetryItemFromJson(json);
+        }
+
+        throw new Exception("Can't find document type: " + docType.toString());
+    }
+
+    /**
+     * Converts JSON object to Request TelemetryItem
+     * @param json The JSON object
+     * @return A TelemetryItem
+     */
+    private TelemetryItem getRequestTelemetryItemFromJson(JSONObject json) throws URISyntaxException, JSONException {
+        System.out.println("Converting JSON object to telemetry item with document type " + DocumentType.Requests.toString());
+        String address       = json.getJSONArray("request").getJSONObject(0).getString("url");
+        Integer port         = json.getJSONArray("request").getJSONObject(0).getJSONObject("urlData").getInt("port");
+        Integer responseCode = json.getJSONArray("request").getJSONObject(0).getInt("responseCode");
+        String userAgent     = json.getJSONObject("context").getJSONObject("device").getString("userAgent");
+
+        JSONArray parameters = json.getJSONArray("request").getJSONObject(0).getJSONObject("urlData").getJSONArray("queryParameters");
+        Hashtable<String, String> queryParameters = new Hashtable<String, String>();
+        for (int i = 0; i < parameters.length(); ++i) {
+            String name  = parameters.getJSONObject(i).getString("parameter");
+            String value = parameters.getJSONObject(i).getString("value");
+            queryParameters.put(name, value);
+        }
+
+        TelemetryItem telemetryResult = new TelemetryItem(DocumentType.Requests, queryParameters.get("requestId"));
+        telemetryResult.setProperty("uri", address);
+        telemetryResult.setProperty("port", port.toString());
+        telemetryResult.setProperty("responseCode", responseCode.toString());
+        telemetryResult.setProperty("userAgent", userAgent);
+
+        for (String key : queryParameters.keySet()) {
+            telemetryResult.setProperty("queryParameter." + key, queryParameters.get(key));
+        }
+
+        return telemetryResult;
+    }
+
+    /**
+     * Converts JSON object to PerformanceCounter TelemetryItem
+     * @param json The JSON object
+     * @return A TelemetryItem
+     */
+    private TelemetryItem getPerformanceCounterTelemetryItemFromJson(JSONObject json) throws JSONException {
+        System.out.println("Converting JSON object to telemetry item with document type " + DocumentType.PerformanceCounters.toString());
+        String category = json.getJSONArray("performanceCounter").getJSONObject(0).getString("categoryName");
```

Comment:
Please use a local variable for `json.getJSONArray("performanceCounter").getJSONObject(0)`